test/dynamo/test_subclasses.py

```diff
@@ -1696,6 +1871,664 @@ def fn(t0, t1, t2):
+        b = a.clone()
+        tt = AnotherTwoTensor(a, b)
+
+        fw, bw = self._compile_check(f, [(tt,)], dynamic=True, call_backward=True)
+
+        self.assertExpectedInline(
+            normalize_gm(fw[0].print_readable(print_output=False)),
+            """\
+class GraphModule(torch.nn.Module):
+    def forward(self, primals_1: "f32[s0, s1]", primals_2: "f32[s0, s1]", primals_3: "Sym(s0)", primals_4: "Sym(s1)", primals_5: "Sym(s0)", primals_6: "Sym(s1)"):
+        clone: "f32[s0, s1]" = torch.ops.aten.clone.default(primals_1);  primals_1 = None
+        clone_1: "f32[s0, s1]" = torch.ops.aten.clone.default(primals_2);  primals_2 = None
+
+        mul: "Sym(s0*s1)" = primals_5 * primals_6
+        view: "f32[s0*s1]" = torch.ops.aten.view.default(clone, [mul])
+        view_1: "f32[s0*s1]" = torch.ops.aten.view.default(clone_1, [mul]);  clone_1 = None
+        return (clone, view, view_1, mul, primals_5, primals_6)
+""",  # noqa: B950
+        )
+
+        self.assertExpectedInline(
+            normalize_gm(bw[0].print_readable(print_output=False)),
+            """\
+class GraphModule(torch.nn.Module):
+    def forward(self, primals_5: "Sym(s0)", primals_6: "Sym(s1)", tangents_1: "f32[s0*s1]", tangents_2: "f32[s0*s1]"):
+        view_2: "f32[s0, s1]" = torch.ops.aten.view.default(tangents_1, [primals_5, primals_6]);  tangents_1 = None
+        view_3: "f32[s0, s1]" = torch.ops.aten.view.default(tangents_2, [primals_5, primals_6]);  tangents_2 = None
+        return (view_2, view_3, None, None, primals_5, primals_6)
+""",  # noqa: B950
+        )
+
+    @unittest.expectedFailure
+    def test_tensor_subclass_TwoTensor_return_multiple(self):
+        def f(tt):
+            y = tt.clone()
+            z = tt.clone()
+            return y.a, y.view(y.shape[0] * y.shape[1]), y.b, z.view(-1)
+
+        a = torch.ones(3, 4, requires_grad=True)
+        b = a.clone()
+        tt = AnotherTwoTensor(a, b)
+
+        fw, bw = self._compile_check(f, [(tt,)], dynamic=True, call_backward=True)
+
+        self.assertExpectedInline(
+            normalize_gm(fw[0].print_readable(print_output=False)),
+            """\
+class GraphModule(torch.nn.Module):
+    def forward(self, primals_1: "f32[3, 4]", primals_2: "f32[3, 4]", primals_3: "Sym(3)", primals_4: "Sym(4)", primals_5: "Sym(3)", primals_6: "Sym(4)"):
+        clone: "f32[3, 4]" = torch.ops.aten.clone.default(primals_1);  primals_1 = None
+        clone_1: "f32[3, 4]" = torch.ops.aten.clone.default(primals_2);  primals_2 = None
+
+        mul: "Sym(12)" = primals_5 * primals_6
+        view: "f32[12]" = torch.ops.aten.view.default(clone, [mul])
+        view_1: "f32[12]" = torch.ops.aten.view.default(clone_1, [mul]);  clone_1 = None
+        return [clone, view, view_1, mul, primals_5, primals_6]
+""",  # noqa: B950
+        )
+
+        self.assertExpectedInline(
+            normalize_gm(bw[0].print_readable(print_output=False)),
+            """\
+class GraphModule(torch.nn.Module):
```

Comment:
I've found expecttests pretty nice for:
(1) ensuring that we don't regress over time if we care about the exact format of the graph
(2) they can help you understand what compile is doing by just looking at an example test and what it's expected graph is

Although when there are a lot of them in a test file it can be a bit unwieldy. Do you think asserting on the expected graph on *all* of these tests provides extra value over just asserting correctness of the output? I would tentatively vote to only keep the expecttest assertions around for a subset. Something like:
(1) basic test showing the graph, where we see extra symint graph inputs/outputs corresponding to dynamic size of subclass input/output
(2) another one but showing actual size compute happening on those symints in the graph, and/or the symints being saved for backward in the case of training

Reply:
Maybe we don't need for all tests. But I was able to catch a few failures by looking at the generated graph.

Reply:
Unless something big change, the generated graph should be stable, no?